python/caffe/io.py

```diff
@@ -20,11 +20,20 @@ def blobproto_to_array(blob, return_diff=False):
     Convert a blob proto to an array. In default, we will just return the data,
     unless return_diff is True, in which case we will return the diff.
     """
+    # Read the data into an array
     if return_diff:
-        return np.array(blob.diff).reshape(*blob.shape.dim)
+        data = np.array(blob.diff)
     else:
-        return np.array(blob.data).reshape(*blob.shape.dim)
+        data = np.array(blob.data)
 
+    if blob.HasField('shape'):
+        return data.reshape(blob.shape.dim)
+    elif blob.HasField('num') or blob.HasField('channels') or blob.HasField('height') or blob.HasField('width'):
+        # Use legacy 4D shape
+        return data.reshape(blob.num, blob.channels, blob.height, blob.width)
+    else:
+        # Return scalar
+        return data
```

Comment:
Well I'm not letting it fall back to this worthless error:
```
  File "/home/tranlaman/BLVC-caffe/python/caffe/io.py", line 26, in blobproto_to_array
    return np.array(blob.data).reshape(*blob.shape.dim)
TypeError: function takes exactly 1 argument (0 given)
```
How about raising a `ValueError`?
```python
          if len(data) != 1:
              raise ValueError('blob has no shape')
          # Return scalar
          return data[0]
```